### PR TITLE
:sparkles: Rework of the search function in swap components

### PR DESCRIPTION
**Pull request by pabloalba**
Implements https://tree.taiga.io/project/penpot/issue/6028

### CODE REVIEW

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs [hunk 1]
```diff
@@ -149,7 +149,7 @@
 
 (mf/defc component-swap-item
   {::mf/wrap-props false}
-  [{:keys [item loop? shapes file-id root-shape container component-id] :as props}]
+  [{:keys [item loop? shapes file-id root-shape container component-id is-search?] :as props}]
```
Comment (niwinz) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs]: Try to no use `?` in components props

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs [hunk 2]
```diff
@@ -212,36 +212,40 @@
...
+                              (let [split-path (cph/split-path path)]
+                                (reduce (fn [acc dir]
+                                          (conj acc (str (last acc) " / " dir)))
+                                        [(first split-path)] (rest split-path))))
+
+        groups              (when-not is-search?
+                              (->> components
+                                   (map :path)
+                                   (map get-subgroups)
+                                   (apply concat)
```
Comment (niwinz) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs]: map + apply concat -> `mapcat`

Also: consider using transducers here.

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs (hunk 1 above)
Comment (pabloalba) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs]: Fixed

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs (hunk 2 above)
Comment (pabloalba) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs]: Fixed

**Pull request merged**